prime-router/src/main/kotlin/azure/BatchFunction.kt

```diff
@@ -51,10 +51,14 @@ class BatchFunction {
                     actionHistory.trackExistingInputReport(it.task.reportId)
                     report
                 }
-                val outReports = when (receiver.batch?.operation) {
+                val mergedReports = when (receiver.batch?.operation) {
                     OrganizationService.BatchOperation.MERGE -> listOf(Report.merge(inReports))
                     else -> inReports
                 }
+                val outReports = when (receiver.format) {
+                    OrganizationService.Format.HL7 -> mergedReports.flatMap { it.split() }
+                    else -> mergedReports
+                }
```

Comment:
I'm confused by this split here.   It looks like the split() is occurring in ReportFunction, so why do it here?   Splitting here is really confusing - it means that, in some cases, BatchFunction does the _opposite_ of batching.

A cleaner solution might be to allow SendFunction to optionally run on a cron timer, just like BatchFunction does now.  Then the workflow for TX might be like this:
1.   ReportFunction creates separate split hl7 files for each TX test result.
2. Skip the BatchFunction step completely, and queue a Send request at a specific time.
3. Send wakes up at the desired TX time, and sends all the accumulated HL7 files to TX.

Reply:
Again we have a different view of what the batch function is for. Send is the function that does retries. Batch is for time-based work.